### ISSUE TITLE
feat: resolve foreignIdFor() column type from referenced model's primary key

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#521]
## Description

`foreignIdFor(SomeModel::class)` always produces `int` type, even when the referenced model uses a UUID or ULID primary key. This causes false positives when the FK column is actually a `string`.

## Current behavior

```php
$table->foreignIdFor(UuidModel::class); // → SchemaColumn('uuid_model_id', 'int')
```

## Expected behavior

```php
$table->foreignIdFor(UuidModel::class); // → SchemaColumn('uuid_model_id', 'string') if PK is uuid/ulid
```

## Proposed solution

After resolving the column name via `getForeignKey()`, look up the referenced model's primary key type in the already-parsed schema:

1. Get the model's table name via `getTable()`
2. Look up the PK column (usually `id`) in `$this->tables[$tableName]`
3. Use the PK column's type for the FK column

Larastan implements this in `SchemaAggregator::getModelReferenceType()` — instantiates the model, gets its table, looks up the `id` column type from the parsed schema.

## References

- Larastan implementation: `SchemaAggregator.php` lines 412-436
